build/build-in-docker

```diff
@@ -24,30 +24,39 @@ set -e
 SCRIPTDIR=$(cd $(dirname $0); pwd)
 
 LOCAL_MAVEN_REPO=${LOCAL_MAVEN_REPO:-"$HOME/.m2/repository"}
-CUDF_USE_PER_THREAD_DEFAULT_STREAM=${CUDF_USE_PER_THREAD_DEFAULT_STREAM:-ON}
 USE_GDS=${USE_GDS:-ON}
 export CMAKE_GENERATOR=${CMAKE_GENERATOR:-"Ninja"}
+export DOCKER_BUILD_EXTRA_ARGS="--platform=linux/amd64 --build-arg CMAKE_ARCH=x86_64"
+CUDA_VER=${CUDA_VER:-cuda11}
+USE_SANITIZER=${USE_SANITIZER:-ON}
+BUILD_FAULTINJ=${BUILD_FAULTINJ:-ON}
 
 if (( $# == 0 )); then
   echo "Usage: $0 <Maven build arguments>"
   exit 1
 fi
 
-_CUDF_CLEAN_SKIP=""
-# if ccache is enabled and libcudf.clean.skip not provided
-# by the user remove the cpp build directory
-#
-if [[ "$CCACHE_DISABLE" != "1" ]]; then
-  if [[ ! "$*" =~ " -Dlibcudf.clean.skip=" ]]; then
-    # Don't skip clean if ccache is enabled
-    # unless the user overrides
-    _CUDF_CLEAN_SKIP="-Dlibcudf.clean.skip=false"
-  fi
+case $(uname -m) in
+  x86_64|amd64)
+    arch=amd64;;
+  aarch64|arm64)
+    arch=arm64;;
+  *)
+    echo "Unsupported CPU architecture"; exit 1;;
+esac
+
+# Set env for arm64 build
+if [ "$arch" == "arm64" ]; then
+  profiles="${profiles},arm64"
+  USE_GDS="OFF"
+  USE_SANITIZER="ON"
+  BUILD_FAULTINJ="OFF"
+  export DOCKER_BUILD_EXTRA_ARGS="--platform=linux/arm64 --build-arg CMAKE_ARCH=aarch64"
 fi
 
 $SCRIPTDIR/run-in-docker mvn \
     -Dmaven.repo.local=$LOCAL_MAVEN_REPO \
-    -DCUDF_USE_PER_THREAD_DEFAULT_STREAM=$CUDF_USE_PER_THREAD_DEFAULT_STREAM \
     -DUSE_GDS=$USE_GDS \
-    $_CUDF_CLEAN_SKIP \
+    -DBUILD_TESTS=ON -DBUILD_FAULTINJ=${BUILD_FAULTINJ} -Dcuda.version=$CUDA_VER \
+    -DUSE_SANITIZER=${USE_SANITIZER} \
```

Comment:
As mentioned before, this script is used by developers, shouldn't be doing surprising things like adding cmdline options to Maven that may conflict with options the developer is specifying.  Not sure why these aren't specified in the CI jobs using this script rather than putting them here.  If we want these to always be on, they shouldn't be specified here, they should be specified in the pom.xml.  Specifically talking about `BUILD_TESTS`,  and `USE_SANITIZER`.  I'm OK with specifying cuda.version since that's closely tied to the Docker image used, and run-in-docker is already deciding that based on CUDA_VERSION.